// @cartesi/token/dist/src/types/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SimpleFaucet__factory = exports.MineTimelock__factory = exports.CartesiTokenTimelock__factory = exports.CartesiToken__factory = exports.TokenTimelock__factory = exports.IERC20__factory = exports.ERC20Mintable__factory = exports.ERC20Detailed__factory = exports.ERC20__factory = exports.Ownable__factory = exports.Context__factory = exports.MinterRole__factory = exports.factories = void 0;
exports.factories = __importStar(require("./factories"));
var MinterRole__factory_1 = require("./factories/@openzeppelin/contracts/access/roles/MinterRole__factory");
Object.defineProperty(exports, "MinterRole__factory", { enumerable: true, get: function () { return MinterRole__factory_1.MinterRole__factory; } });
var Context__factory_1 = require("./factories/@openzeppelin/contracts/GSN/Context__factory");
Object.defineProperty(exports, "Context__factory", { enumerable: true, get: function () { return Context__factory_1.Context__factory; } });
var Ownable__factory_1 = require("./factories/@openzeppelin/contracts/ownership/Ownable__factory");
Object.defineProperty(exports, "Ownable__factory", { enumerable: true, get: function () { return Ownable__factory_1.Ownable__factory; } });
var ERC20__factory_1 = require("./factories/@openzeppelin/contracts/token/ERC20/ERC20__factory");
Object.defineProperty(exports, "ERC20__factory", { enumerable: true, get: function () { return ERC20__factory_1.ERC20__factory; } });
var ERC20Detailed__factory_1 = require("./factories/@openzeppelin/contracts/token/ERC20/ERC20Detailed__factory");
Object.defineProperty(exports, "ERC20Detailed__factory", { enumerable: true, get: function () { return ERC20Detailed__factory_1.ERC20Detailed__factory; } });
var ERC20Mintable__factory_1 = require("./factories/@openzeppelin/contracts/token/ERC20/ERC20Mintable__factory");
Object.defineProperty(exports, "ERC20Mintable__factory", { enumerable: true, get: function () { return ERC20Mintable__factory_1.ERC20Mintable__factory; } });
var IERC20__factory_1 = require("./factories/@openzeppelin/contracts/token/ERC20/IERC20__factory");
Object.defineProperty(exports, "IERC20__factory", { enumerable: true, get: function () { return IERC20__factory_1.IERC20__factory; } });
var TokenTimelock__factory_1 = require("./factories/@openzeppelin/contracts/token/ERC20/TokenTimelock__factory");
Object.defineProperty(exports, "TokenTimelock__factory", { enumerable: true, get: function () { return TokenTimelock__factory_1.TokenTimelock__factory; } });
var CartesiToken__factory_1 = require("./factories/contracts/CartesiToken__factory");
Object.defineProperty(exports, "CartesiToken__factory", { enumerable: true, get: function () { return CartesiToken__factory_1.CartesiToken__factory; } });
var CartesiTokenTimelock__factory_1 = require("./factories/contracts/CartesiTokenTimeLock.sol/CartesiTokenTimelock__factory");
Object.defineProperty(exports, "CartesiTokenTimelock__factory", { enumerable: true, get: function () { return CartesiTokenTimelock__factory_1.CartesiTokenTimelock__factory; } });
var MineTimelock__factory_1 = require("./factories/contracts/MineTimelock__factory");
Object.defineProperty(exports, "MineTimelock__factory", { enumerable: true, get: function () { return MineTimelock__factory_1.MineTimelock__factory; } });
var SimpleFaucet__factory_1 = require("./factories/contracts/SimpleFaucet__factory");
Object.defineProperty(exports, "SimpleFaucet__factory", { enumerable: true, get: function () { return SimpleFaucet__factory_1.SimpleFaucet__factory; } });
