// @cartesi/rollups/dist/src/types/factories/contracts/test_helper/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TestLibClaimsMask__factory = exports.SimpleToken__factory = exports.SimpleNFT__factory = exports.SimpleContract__factory = exports.upgradeInitializers = exports.facets = void 0;
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
exports.facets = __importStar(require("./facets"));
exports.upgradeInitializers = __importStar(require("./upgrade_initializers"));
var SimpleContract__factory_1 = require("./SimpleContract__factory");
Object.defineProperty(exports, "SimpleContract__factory", { enumerable: true, get: function () { return SimpleContract__factory_1.SimpleContract__factory; } });
var SimpleNFT__factory_1 = require("./SimpleNFT__factory");
Object.defineProperty(exports, "SimpleNFT__factory", { enumerable: true, get: function () { return SimpleNFT__factory_1.SimpleNFT__factory; } });
var SimpleToken__factory_1 = require("./SimpleToken__factory");
Object.defineProperty(exports, "SimpleToken__factory", { enumerable: true, get: function () { return SimpleToken__factory_1.SimpleToken__factory; } });
var TestLibClaimsMask__factory_1 = require("./TestLibClaimsMask__factory");
Object.defineProperty(exports, "TestLibClaimsMask__factory", { enumerable: true, get: function () { return TestLibClaimsMask__factory_1.TestLibClaimsMask__factory; } });
