// @cartesi/rollups/dist/src/types/factories/contracts/libraries/LibFeeManager__factory.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LibFeeManager__factory = void 0;
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
const ethers_1 = require("ethers");
const _abi = [
    {
        anonymous: false,
        inputs: [
            {
                indexed: false,
                internalType: "uint256",
                name: "value",
                type: "uint256",
            },
        ],
        name: "FeePerClaimReset",
        type: "event",
    },
    {
        anonymous: false,
        inputs: [
            {
                indexed: false,
                internalType: "address",
                name: "validator",
                type: "address",
            },
            {
                indexed: false,
                internalType: "uint256",
                name: "claims",
                type: "uint256",
            },
        ],
        name: "FeeRedeemed",
        type: "event",
    },
];
const _bytecode = "0x60566037600b82828239805160001a607314602a57634e487b7160e01b600052600060045260246000fd5b30600052607381538281f3fe73000000000000000000000000000000000000000030146080604052600080fdfea2646970667358221220d4bfde1ca899685135034a94683fa3818f6fc5a557a7e65a715f26e8c474375664736f6c634300080d0033";
const isSuperArgs = (xs) => xs.length > 1;
class LibFeeManager__factory extends ethers_1.ContractFactory {
    constructor(...args) {
        if (isSuperArgs(args)) {
            super(...args);
        }
        else {
            super(_abi, _bytecode, args[0]);
        }
    }
    deploy(overrides) {
        return super.deploy(overrides || {});
    }
    getDeployTransaction(overrides) {
        return super.getDeployTransaction(overrides || {});
    }
    attach(address) {
        return super.attach(address);
    }
    connect(signer) {
        return super.connect(signer);
    }
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.LibFeeManager__factory = LibFeeManager__factory;
LibFeeManager__factory.bytecode = _bytecode;
LibFeeManager__factory.abi = _abi;
