// @cartesi/rollups/dist/src/types/factories/contracts/libraries/LibRollups__factory.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LibRollups__factory = void 0;
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
const ethers_1 = require("ethers");
const _abi = [
    {
        anonymous: false,
        inputs: [
            {
                indexed: true,
                internalType: "uint256",
                name: "epochNumber",
                type: "uint256",
            },
            {
                indexed: false,
                internalType: "bytes32",
                name: "epochHash",
                type: "bytes32",
            },
        ],
        name: "FinalizeEpoch",
        type: "event",
    },
    {
        anonymous: false,
        inputs: [
            {
                indexed: false,
                internalType: "enum Phase",
                name: "newPhase",
                type: "uint8",
            },
        ],
        name: "PhaseChange",
        type: "event",
    },
    {
        anonymous: false,
        inputs: [
            {
                indexed: false,
                internalType: "address",
                name: "winner",
                type: "address",
            },
            {
                indexed: false,
                internalType: "address",
                name: "loser",
                type: "address",
            },
            {
                indexed: false,
                internalType: "bytes32",
                name: "winningClaim",
                type: "bytes32",
            },
        ],
        name: "ResolveDispute",
        type: "event",
    },
];
const _bytecode = "0x60566037600b82828239805160001a607314602a57634e487b7160e01b600052600060045260246000fd5b30600052607381538281f3fe73000000000000000000000000000000000000000030146080604052600080fdfea264697066735822122037931b7a7a765b62a3eda86e3ed16ca1dff5949bf3344ac25376cd1ebf6a7c3264736f6c634300080d0033";
const isSuperArgs = (xs) => xs.length > 1;
class LibRollups__factory extends ethers_1.ContractFactory {
    constructor(...args) {
        if (isSuperArgs(args)) {
            super(...args);
        }
        else {
            super(_abi, _bytecode, args[0]);
        }
    }
    deploy(overrides) {
        return super.deploy(overrides || {});
    }
    getDeployTransaction(overrides) {
        return super.getDeployTransaction(overrides || {});
    }
    attach(address) {
        return super.attach(address);
    }
    connect(signer) {
        return super.connect(signer);
    }
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.LibRollups__factory = LibRollups__factory;
LibRollups__factory.bytecode = _bytecode;
LibRollups__factory.abi = _abi;
