// @cartesi/rollups/dist/src/types/factories/contracts/libraries/alternatives/LibValidatorManager1__factory.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LibValidatorManager1__factory = void 0;
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
const ethers_1 = require("ethers");
const _abi = [
    {
        anonymous: false,
        inputs: [
            {
                indexed: false,
                internalType: "enum Result",
                name: "result",
                type: "uint8",
            },
            {
                indexed: false,
                internalType: "bytes32[2]",
                name: "claims",
                type: "bytes32[2]",
            },
            {
                indexed: false,
                internalType: "address payable[2]",
                name: "validators",
                type: "address[2]",
            },
        ],
        name: "ClaimReceived",
        type: "event",
    },
    {
        anonymous: false,
        inputs: [
            {
                indexed: false,
                internalType: "enum Result",
                name: "result",
                type: "uint8",
            },
            {
                indexed: false,
                internalType: "bytes32[2]",
                name: "claims",
                type: "bytes32[2]",
            },
            {
                indexed: false,
                internalType: "address payable[2]",
                name: "validators",
                type: "address[2]",
            },
        ],
        name: "DisputeEnded",
        type: "event",
    },
    {
        anonymous: false,
        inputs: [
            {
                indexed: false,
                internalType: "bytes32",
                name: "claim",
                type: "bytes32",
            },
        ],
        name: "NewEpoch",
        type: "event",
    },
];
const _bytecode = "0x60566037600b82828239805160001a607314602a57634e487b7160e01b600052600060045260246000fd5b30600052607381538281f3fe73000000000000000000000000000000000000000030146080604052600080fdfea26469706673582212203dd2bc7c2a7b765453aa79adbdad73434fb1913566d7984a24a18a58d5da304064736f6c634300080d0033";
const isSuperArgs = (xs) => xs.length > 1;
class LibValidatorManager1__factory extends ethers_1.ContractFactory {
    constructor(...args) {
        if (isSuperArgs(args)) {
            super(...args);
        }
        else {
            super(_abi, _bytecode, args[0]);
        }
    }
    deploy(overrides) {
        return super.deploy(overrides || {});
    }
    getDeployTransaction(overrides) {
        return super.getDeployTransaction(overrides || {});
    }
    attach(address) {
        return super.attach(address);
    }
    connect(signer) {
        return super.connect(signer);
    }
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.LibValidatorManager1__factory = LibValidatorManager1__factory;
LibValidatorManager1__factory.bytecode = _bytecode;
LibValidatorManager1__factory.abi = _abi;
