// @cartesi/rollups/dist/src/types/factories/contracts/facets/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ValidatorManagerFacet__factory = exports.RollupsFacet__factory = exports.OwnershipFacet__factory = exports.OutputFacet__factory = exports.InputFacet__factory = exports.FeeManagerFacet__factory = exports.EtherPortalFacet__factory = exports.ERC721PortalFacet__factory = exports.ERC20PortalFacet__factory = exports.DiamondLoupeFacet__factory = exports.DiamondCutFacet__factory = exports.DebugFacet__factory = exports.alternatives = void 0;
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
exports.alternatives = __importStar(require("./alternatives"));
var DebugFacet__factory_1 = require("./DebugFacet__factory");
Object.defineProperty(exports, "DebugFacet__factory", { enumerable: true, get: function () { return DebugFacet__factory_1.DebugFacet__factory; } });
var DiamondCutFacet__factory_1 = require("./DiamondCutFacet__factory");
Object.defineProperty(exports, "DiamondCutFacet__factory", { enumerable: true, get: function () { return DiamondCutFacet__factory_1.DiamondCutFacet__factory; } });
var DiamondLoupeFacet__factory_1 = require("./DiamondLoupeFacet__factory");
Object.defineProperty(exports, "DiamondLoupeFacet__factory", { enumerable: true, get: function () { return DiamondLoupeFacet__factory_1.DiamondLoupeFacet__factory; } });
var ERC20PortalFacet__factory_1 = require("./ERC20PortalFacet__factory");
Object.defineProperty(exports, "ERC20PortalFacet__factory", { enumerable: true, get: function () { return ERC20PortalFacet__factory_1.ERC20PortalFacet__factory; } });
var ERC721PortalFacet__factory_1 = require("./ERC721PortalFacet__factory");
Object.defineProperty(exports, "ERC721PortalFacet__factory", { enumerable: true, get: function () { return ERC721PortalFacet__factory_1.ERC721PortalFacet__factory; } });
var EtherPortalFacet__factory_1 = require("./EtherPortalFacet__factory");
Object.defineProperty(exports, "EtherPortalFacet__factory", { enumerable: true, get: function () { return EtherPortalFacet__factory_1.EtherPortalFacet__factory; } });
var FeeManagerFacet__factory_1 = require("./FeeManagerFacet__factory");
Object.defineProperty(exports, "FeeManagerFacet__factory", { enumerable: true, get: function () { return FeeManagerFacet__factory_1.FeeManagerFacet__factory; } });
var InputFacet__factory_1 = require("./InputFacet__factory");
Object.defineProperty(exports, "InputFacet__factory", { enumerable: true, get: function () { return InputFacet__factory_1.InputFacet__factory; } });
var OutputFacet__factory_1 = require("./OutputFacet__factory");
Object.defineProperty(exports, "OutputFacet__factory", { enumerable: true, get: function () { return OutputFacet__factory_1.OutputFacet__factory; } });
var OwnershipFacet__factory_1 = require("./OwnershipFacet__factory");
Object.defineProperty(exports, "OwnershipFacet__factory", { enumerable: true, get: function () { return OwnershipFacet__factory_1.OwnershipFacet__factory; } });
var RollupsFacet__factory_1 = require("./RollupsFacet__factory");
Object.defineProperty(exports, "RollupsFacet__factory", { enumerable: true, get: function () { return RollupsFacet__factory_1.RollupsFacet__factory; } });
var ValidatorManagerFacet__factory_1 = require("./ValidatorManagerFacet__factory");
Object.defineProperty(exports, "ValidatorManagerFacet__factory", { enumerable: true, get: function () { return ValidatorManagerFacet__factory_1.ValidatorManagerFacet__factory; } });
