// @cartesi/rollups/dist/src/types/factories/contracts/test_helper/upgrade_initializers/DS6Downgrade__factory.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DS6Downgrade__factory = void 0;
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [],
        name: "downgrade",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
];
const _bytecode = "0x6080604052348015600f57600080fd5b50608d8061001e6000396000f3fe6080604052348015600f57600080fd5b506004361060285760003560e01c806306ce466414602d575b600080fd5b605560007fe44d9aca4747abb7268e5218fc7ac0390858570cb5a7d6dc5923e710e5a5ddcc55565b00fea26469706673582212203630b77c08ed2d26ffc473a14c04ec40222e074524aa16b27529c757246e991c64736f6c634300080d0033";
const isSuperArgs = (xs) => xs.length > 1;
class DS6Downgrade__factory extends ethers_1.ContractFactory {
    constructor(...args) {
        if (isSuperArgs(args)) {
            super(...args);
        }
        else {
            super(_abi, _bytecode, args[0]);
        }
    }
    deploy(overrides) {
        return super.deploy(overrides || {});
    }
    getDeployTransaction(overrides) {
        return super.getDeployTransaction(overrides || {});
    }
    attach(address) {
        return super.attach(address);
    }
    connect(signer) {
        return super.connect(signer);
    }
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.DS6Downgrade__factory = DS6Downgrade__factory;
DS6Downgrade__factory.bytecode = _bytecode;
DS6Downgrade__factory.abi = _abi;
