// @cartesi/rollups/dist/src/types/factories/contracts/test_helper/upgrade_initializers/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DS9Upgrade__factory = exports.DS7Upgrade__factory = exports.DS7Init__factory = exports.DS7Downgrade__factory = exports.DS6Upgrade__factory = exports.DS6Init__factory = exports.DS6Downgrade__factory = exports.DS4Upgrade__factory = exports.DS3Upgrade__factory = exports.DS2Upgrade__factory = exports.DS1Init__factory = void 0;
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
var DS1Init__factory_1 = require("./DS1Init__factory");
Object.defineProperty(exports, "DS1Init__factory", { enumerable: true, get: function () { return DS1Init__factory_1.DS1Init__factory; } });
var DS2Upgrade__factory_1 = require("./DS2Upgrade__factory");
Object.defineProperty(exports, "DS2Upgrade__factory", { enumerable: true, get: function () { return DS2Upgrade__factory_1.DS2Upgrade__factory; } });
var DS3Upgrade__factory_1 = require("./DS3Upgrade__factory");
Object.defineProperty(exports, "DS3Upgrade__factory", { enumerable: true, get: function () { return DS3Upgrade__factory_1.DS3Upgrade__factory; } });
var DS4Upgrade__factory_1 = require("./DS4Upgrade__factory");
Object.defineProperty(exports, "DS4Upgrade__factory", { enumerable: true, get: function () { return DS4Upgrade__factory_1.DS4Upgrade__factory; } });
var DS6Downgrade__factory_1 = require("./DS6Downgrade__factory");
Object.defineProperty(exports, "DS6Downgrade__factory", { enumerable: true, get: function () { return DS6Downgrade__factory_1.DS6Downgrade__factory; } });
var DS6Init__factory_1 = require("./DS6Init__factory");
Object.defineProperty(exports, "DS6Init__factory", { enumerable: true, get: function () { return DS6Init__factory_1.DS6Init__factory; } });
var DS6Upgrade__factory_1 = require("./DS6Upgrade__factory");
Object.defineProperty(exports, "DS6Upgrade__factory", { enumerable: true, get: function () { return DS6Upgrade__factory_1.DS6Upgrade__factory; } });
var DS7Downgrade__factory_1 = require("./DS7Downgrade__factory");
Object.defineProperty(exports, "DS7Downgrade__factory", { enumerable: true, get: function () { return DS7Downgrade__factory_1.DS7Downgrade__factory; } });
var DS7Init__factory_1 = require("./DS7Init__factory");
Object.defineProperty(exports, "DS7Init__factory", { enumerable: true, get: function () { return DS7Init__factory_1.DS7Init__factory; } });
var DS7Upgrade__factory_1 = require("./DS7Upgrade__factory");
Object.defineProperty(exports, "DS7Upgrade__factory", { enumerable: true, get: function () { return DS7Upgrade__factory_1.DS7Upgrade__factory; } });
var DS9Upgrade__factory_1 = require("./DS9Upgrade__factory");
Object.defineProperty(exports, "DS9Upgrade__factory", { enumerable: true, get: function () { return DS9Upgrade__factory_1.DS9Upgrade__factory; } });
