// @cartesi/rollups/dist/deploy/02_facets.js
"use strict";
// Copyright 2022 Cartesi Pte. Ltd.
Object.defineProperty(exports, "__esModule", { value: true });
const func = async (hre) => {
    const { deployments, getNamedAccounts } = hre;
    const { deployer } = await getNamedAccounts();
    const opts = {
        deterministicDeployment: true,
        from: deployer,
        log: true,
    };
    const Bitmask = await deployments.get("Bitmask");
    const MerkleV2 = await deployments.get("MerkleV2");
    await deployments.deploy("ERC20PortalFacet", opts);
    await deployments.deploy("ERC721PortalFacet", opts);
    await deployments.deploy("EtherPortalFacet", opts);
    await deployments.deploy("FeeManagerFacet", opts);
    await deployments.deploy("InputFacet", opts);
    await deployments.deploy("OutputFacet", {
        ...opts,
        libraries: {
            Bitmask: Bitmask.address,
            MerkleV2: MerkleV2.address,
        },
    });
    await deployments.deploy("RollupsFacet", opts);
    await deployments.deploy("ValidatorManagerFacet", opts);
};
exports.default = func;
func.tags = ["RollupsFacets"];
