// @cartesi/rollups/dist/deploy/05_factory.js
"use strict";
// Copyright 2022 Cartesi Pte. Ltd.
Object.defineProperty(exports, "__esModule", { value: true });
const utils_1 = require("../src/utils");
const func = async (hre) => {
    const { deployments, ethers, getNamedAccounts } = hre;
    const { deployer } = await getNamedAccounts();
    const opts = {
        deterministicDeployment: true,
        from: deployer,
        log: true,
    };
    const { DiamondCutFacet, DiamondInit, Bank } = await deployments.all();
    const facetCuts = await (0, utils_1.getFacetCuts)(utils_1.productionFacetNames);
    let factoryConfig = {
        diamondCutFacet: DiamondCutFacet.address,
        diamondInit: DiamondInit.address,
        feeManagerBank: Bank.address,
        diamondCut: facetCuts,
    };
    await deployments.deploy("CartesiDAppFactory", {
        ...opts,
        args: [factoryConfig],
    });
};
exports.default = func;
func.dependencies = ["RollupsFacets", "RollupsInit", "RollupsDiamond", "Bank"];
func.tags = ["RollupsFactory"];
