// @cartesi/rollups/dist/src/types/factories/contracts/interfaces/IValidatorManager__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IValidatorManager__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        anonymous: false,
        inputs: [
            {
                indexed: false,
                internalType: "enum Result",
                name: "result",
                type: "uint8",
            },
            {
                indexed: false,
                internalType: "bytes32[2]",
                name: "claims",
                type: "bytes32[2]",
            },
            {
                indexed: false,
                internalType: "address payable[2]",
                name: "validators",
                type: "address[2]",
            },
        ],
        name: "ClaimReceived",
        type: "event",
    },
    {
        anonymous: false,
        inputs: [
            {
                indexed: false,
                internalType: "enum Result",
                name: "result",
                type: "uint8",
            },
            {
                indexed: false,
                internalType: "bytes32[2]",
                name: "claims",
                type: "bytes32[2]",
            },
            {
                indexed: false,
                internalType: "address payable[2]",
                name: "validators",
                type: "address[2]",
            },
        ],
        name: "DisputeEnded",
        type: "event",
    },
    {
        anonymous: false,
        inputs: [
            {
                indexed: false,
                internalType: "bytes32",
                name: "claim",
                type: "bytes32",
            },
        ],
        name: "NewEpoch",
        type: "event",
    },
    {
        inputs: [],
        name: "getCurrentClaim",
        outputs: [
            {
                internalType: "bytes32",
                name: "",
                type: "bytes32",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
class IValidatorManager__factory {
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.IValidatorManager__factory = IValidatorManager__factory;
IValidatorManager__factory.abi = _abi;
