// @cartesi/rollups/dist/src/types/factories/contracts/test_helper/facets/Foo2Facet__factory.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Foo2Facet__factory = void 0;
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [],
        name: "foo",
        outputs: [
            {
                internalType: "uint256",
                name: "",
                type: "uint256",
            },
        ],
        stateMutability: "pure",
        type: "function",
    },
];
const _bytecode = "0x6080604052348015600f57600080fd5b50607780601d6000396000f3fe6080604052348015600f57600080fd5b506004361060285760003560e01c8063c298557814602d575b600080fd5b600a60405190815260200160405180910390f3fea26469706673582212202d70abb3995117c4c350758d9066cdbd4d9630f307936b361fa08796dc0aa73064736f6c634300080d0033";
const isSuperArgs = (xs) => xs.length > 1;
class Foo2Facet__factory extends ethers_1.ContractFactory {
    constructor(...args) {
        if (isSuperArgs(args)) {
            super(...args);
        }
        else {
            super(_abi, _bytecode, args[0]);
        }
    }
    deploy(overrides) {
        return super.deploy(overrides || {});
    }
    getDeployTransaction(overrides) {
        return super.getDeployTransaction(overrides || {});
    }
    attach(address) {
        return super.attach(address);
    }
    connect(signer) {
        return super.connect(signer);
    }
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.Foo2Facet__factory = Foo2Facet__factory;
Foo2Facet__factory.bytecode = _bytecode;
Foo2Facet__factory.abi = _abi;
