// @cartesi/rollups/dist/src/types/factories/contracts/test_helper/facets/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Foo2Facet__factory = exports.Foo1Facet__factory = exports.DS9Facet__factory = exports.DS8Facet__factory = exports.DS7Facet__factory = exports.DS6Facet__factory = exports.DS5Facet__factory = exports.DS4Facet__factory = exports.DS3Facet__factory = exports.DS2Facet__factory = exports.DS1Facet__factory = void 0;
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
var DS1Facet__factory_1 = require("./DS1Facet__factory");
Object.defineProperty(exports, "DS1Facet__factory", { enumerable: true, get: function () { return DS1Facet__factory_1.DS1Facet__factory; } });
var DS2Facet__factory_1 = require("./DS2Facet__factory");
Object.defineProperty(exports, "DS2Facet__factory", { enumerable: true, get: function () { return DS2Facet__factory_1.DS2Facet__factory; } });
var DS3Facet__factory_1 = require("./DS3Facet__factory");
Object.defineProperty(exports, "DS3Facet__factory", { enumerable: true, get: function () { return DS3Facet__factory_1.DS3Facet__factory; } });
var DS4Facet__factory_1 = require("./DS4Facet__factory");
Object.defineProperty(exports, "DS4Facet__factory", { enumerable: true, get: function () { return DS4Facet__factory_1.DS4Facet__factory; } });
var DS5Facet__factory_1 = require("./DS5Facet__factory");
Object.defineProperty(exports, "DS5Facet__factory", { enumerable: true, get: function () { return DS5Facet__factory_1.DS5Facet__factory; } });
var DS6Facet__factory_1 = require("./DS6Facet__factory");
Object.defineProperty(exports, "DS6Facet__factory", { enumerable: true, get: function () { return DS6Facet__factory_1.DS6Facet__factory; } });
var DS7Facet__factory_1 = require("./DS7Facet__factory");
Object.defineProperty(exports, "DS7Facet__factory", { enumerable: true, get: function () { return DS7Facet__factory_1.DS7Facet__factory; } });
var DS8Facet__factory_1 = require("./DS8Facet__factory");
Object.defineProperty(exports, "DS8Facet__factory", { enumerable: true, get: function () { return DS8Facet__factory_1.DS8Facet__factory; } });
var DS9Facet__factory_1 = require("./DS9Facet__factory");
Object.defineProperty(exports, "DS9Facet__factory", { enumerable: true, get: function () { return DS9Facet__factory_1.DS9Facet__factory; } });
var Foo1Facet__factory_1 = require("./Foo1Facet__factory");
Object.defineProperty(exports, "Foo1Facet__factory", { enumerable: true, get: function () { return Foo1Facet__factory_1.Foo1Facet__factory; } });
var Foo2Facet__factory_1 = require("./Foo2Facet__factory");
Object.defineProperty(exports, "Foo2Facet__factory", { enumerable: true, get: function () { return Foo2Facet__factory_1.Foo2Facet__factory; } });
